yext/pages
# feat(pages): multibrand setup support in dev and vite-plugin build

This PR adds support for multibrand setup in pages dev and vite-plugin build by exposing a new optional field `domain` in 
`ProjectStructureConfig` to specify the domain of the site, aka the subfolder name inside template root folder and sites config root folder.

As outlined in the spec, this PR updated dev and build logic such that any template files defined in a matching domain subdirectory will be used instead of those with the same name found at the root. Any other additional template files in the template root folder will be shared across the domain specific subdirectory. `features.json` will be generated in root sites-config folder by default, or the domain specific sites-config folder if a domain is provided (will update the feature command in another item).

J=SLAP-2411
TEST=auto&manual

new jest tests for `getTemplatesFilepath` passed

created a [starter repo](https://github.com/yen-tt/pages-starter/tree/dev/test-multibrand) with multibrand setup (`locations.brand1.com` and `locations.brand2.com` folder in site-config and src/templates):
- run `pnpm pack` for yext/pages package with `domain: "locations.brand1.com",` in defaultConfig of `structure.ts`. ran `npm run dev:multibrand` in the repo, see from landing page that location and static page template inside brand1 folder is used AND random and robots template in root folder is used. All pages successfully loaded. ran `npm run build:multibrand` successfully and see the same result.
- run `pnpm pack` for yext/pages package without `domain` in defaultConfig of `structure.ts`. ran `npm run dev` in the repo, see that landing page display links to templates from root folder only, without templates in brand1 or brand2. All pages successfully loaded. ran `npm run build` successfully and see the same result.
